### DISCUSSION TITLE
Generic Higher order Function Confusion

### DISCUSSION THREAD

**Discussion by hmc-cs-mdrissi** [python/typing#1105] [Q&A] [answered]
Why does `set` behave differently from `lambda ls: set(ls)` in the example below?

```python
x = [[1, 2, 3]]
y = map(set, x)
reveal_type(y) # map[set[unknown]]
```

```python
x = [[1, 2, 3]]
y = map(lambda ls: set(ls), x)
reveal_type(y) # map[set[int]]
```

This shows same behavior for both mypy and pyright and I'm confused why the two code blocks are handled differently.

**Comment by erictraut** [comment, marked as answer]
For reference, I answered this question in a related issue in the pyright issue tracker. https://github.com/microsoft/pyright/issues/3208
